File: batch10amnav/SpreasheetExample/app/src/main/java/com/example/spreasheetexample/MyPasswordlist.java
package com.example.spreasheetexample;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.android.volley.DefaultRetryPolicy;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.RetryPolicy;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MyPasswordlist extends AppCompatActivity {
RecyclerView recyclerView;
ArrayList<Item> userlist;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my_passwordlist);
        recyclerView = findViewById(R.id.userrecyclerview);
        userlist =new ArrayList<>();

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext());

        recyclerView.setLayoutManager(linearLayoutManager);
        showuserlist();

    }

    private void showuserlist() {
        StringRequest stringRequest = new StringRequest(Request.Method.GET, "https://script.google.com/macros/s/AKfycbxOQ-A_egwFNSWYD1KJUpXevpPSmGg4ZF3kaWTHKr5e-0p8X5PdgMPFREGTqw2Nzo5j/exec?action=getItems",
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                      parsresponse(response);
                        Toast.makeText(MyPasswordlist.this, response, Toast.LENGTH_SHORT).show();
                    }
                },
                new Response.ErrorListener() {
                    @Override
                    public void onErrorResponse(VolleyError error) {
                        Toast.makeText(getApplicationContext(), "error "+error.getMessage(), Toast.LENGTH_SHORT).show();
                    }
                }
        );

        int soketimeout= 50000;

        RetryPolicy retryPolicy = new DefaultRetryPolicy(soketimeout,0,DefaultRetryPolicy.DEFAULT_BACKOFF_MULT);
        stringRequest.setRetryPolicy(retryPolicy);
        RequestQueue requestQueue = Volley.newRequestQueue(MyPasswordlist.this);

        requestQueue.add(stringRequest);
    }

    private void parsresponse(String response) {


        try {
           JSONObject  jsonObject = new JSONObject(response);
           // JSONArray jsonArray = jobj.getJSONArray("items");
//            for(int i=0;i<jsonArray.length();i++)
//            {
//                JSONObject jo = jsonArray.getJSONObject(i);
//
//                EamilModels student = new EamilModels();
//
//                student.setEmail(jo.getString("email"));
//                student.setNumber(jo.getString("number"));
//                student.setId(jo.getString("id"));
//                student.setDate(jo.getString("date"));
//                arraylist.add(student);
//            }

            JSONArray jsonArray = jsonObject.getJSONArray("items");

            for(int i =0;i<jsonArray.length();i++)
            {
                JSONObject jsonObject1 = jsonArray.getJSONObject(i);
                Item item = new Item();
                item.setUserdate(jsonObject1.getString("userdate"));
                item.setUsername(jsonObject1.getString("username"));
                item.setUserId(jsonObject1.getString("userId"));
                item.setUserpassword(jsonObject1.getString("userpassword"));

                userlist.add(item);

            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

MyPasswordAdapter myPasswordAdapter =new MyPasswordAdapter(userlist,getApplicationContext());
        recyclerView.setAdapter(myPasswordAdapter);

    }
}

File: Rahoul3pm/GoogleSpreasheeExample/app/src/main/java/com/example/googlespreasheeexample/MyPasswordlist.kt
package com.example.googlespreasheeexample

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.*
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import java.util.ArrayList

class MyPasswordlist : AppCompatActivity() {
  lateinit  var recyclerView: RecyclerView
   lateinit var userlist: ArrayList<Item>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my_passwordlist)
        recyclerView = findViewById(R.id.userrecyclerview)
        userlist = ArrayList()
        val linearLayoutManager = LinearLayoutManager(applicationContext)
        recyclerView.layoutManager = linearLayoutManager
        showuserlist()
    }

    private fun showuserlist() {
        val stringRequest = StringRequest(
            Request.Method.GET,
            "https://script.google.com/macros/s/AKfycbxCraFOxZ6tyecjd2V_2XQg-CGJEsUmc0olDVXmtvo_5VOclA94E-Ot3We2be2aMUhbwA/exec?action=getItems",
            { response ->
                parsresponse(response)
                Toast.makeText(this@MyPasswordlist, response, Toast.LENGTH_SHORT).show()
            }
        ) { error ->
            Toast.makeText(
                applicationContext,
                "error " + error.message,
                Toast.LENGTH_SHORT
            ).show()
        }
        val soketimeout = 50000
        val retryPolicy: RetryPolicy =
            DefaultRetryPolicy(soketimeout, 0, DefaultRetryPolicy.DEFAULT_BACKOFF_MULT)
        stringRequest.retryPolicy = retryPolicy
        val requestQueue = Volley.newRequestQueue(this@MyPasswordlist)
        requestQueue.add(stringRequest)
    }

    private fun parsresponse(response: String) {
        try {
            val jsonObject = JSONObject(response)

            val jsonArray = jsonObject.getJSONArray("items")
            for (i in 0 until jsonArray.length()) {
                val jsonObject1 = jsonArray.getJSONObject(i)
                val item = Item()
                item.userdate=jsonObject1.getString("userdate")
                item.username = jsonObject1.getString("username")
                item.userId = jsonObject1.getString("userId")
                item.userpassword =jsonObject1.getString("userpassword")
                userlist.add(item)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        val myPasswordAdapter = MyPasswordAdapter(
            userlist,
            applicationContext
        )
        recyclerView.adapter = myPasswordAdapter
    }
}